example/src/main/java/org/apache/rocketmq/example/benchmark/BatchProducer.java

```diff
@@ -317,7 +320,8 @@ class StatsBenchmarkBatchProducer {
 
     private final LongAdder sendMessageFailedCount = new LongAdder();
 
-    private final Timer timer = new Timer("BenchmarkTimerThread", true);
+    private final ScheduledExecutorService executorService = new ScheduledThreadPoolExecutor(1,
+            new BasicThreadFactory.Builder().namingPattern("BenchmarkTimerThread-%d").daemon(true).build());
```

Comment:
Maybe it's better to use `ThreadFactoryImpl`, just like this:
https://github.com/apache/rocketmq/blob/5d0102396031b728f9accfa5a562fe20906ac424/broker/src/main/java/org/apache/rocketmq/broker/BrokerController.java#L136

Reply:
Emmm, I prefer to like present way